src/modules/user/typedefs/user.ts

```diff
@@ -1,27 +1,54 @@
 import { gql } from "graphql-tag";
 
 const typedef = gql`
+  directive @auth on OBJECT
+
+  """
+  User accout info.
+  """
   type User {
-    id: String!
-    password: String!
-    last_login: String
-    is_superuser: Boolean!
+    email: String!
     username: String!
     first_name: String!
     last_name: String!
-    email: String!
+    major: [String!]!
+    last_login: String!
+    date_joined: String!
     is_staff: Boolean!
     is_active: Boolean!
-    date_joined: String!
-    major: [String],
-    email_class_update: Boolean,
-    email_grade_update: Boolean,
-    email_enrollment_opening: Boolean,
-    email_berkeleytime_update: Boolean
+    email_class_update: Boolean!
+    email_grade_update: Boolean!
+    email_enrollment_opening: Boolean!
+    email_berkeleytime_update: Boolean!
   }
 
-  type Query {
-    User(email: String!): User
+  type Query @auth {
+    """
+    Query for user info.
+    """
+    User: User
+  }
+
+  type Mutation @auth {
+    """
+    Mutate username, first name, and last name.
+    """
+    UpdateUserInfo(username: String, first_name: String, last_name: String): User
+
+    """
+    Mutate major.
+    """
+    UpdateUserMajor(major: [String!]): User
+
+    """
+    Mutate email preferences.
+    """
+    UpdateUserEmailPreferences(email_class_update: Boolean, email_grade_update: Boolean, email_enrollment_opening: Boolean, email_berkeleytime_update: Boolean): User
```

Comment:
Is there a reason why these two mutations are not part of `updateUserInfo`? the parameters are all optional anyway. you can create [input types](https://graphql.org/graphql-js/mutations-and-input-types/) to group the parameters for cleanliness